x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -613,6 +622,19 @@ public void onResponse(Void unused) {
         }
     }
 
+    private IndicesAccessControl wrapIndicesAccessControl(IndicesAccessControl indicesAccessControl) {
+        if (indicesAccessControl == null
+            || indicesAccessControl == IndicesAccessControl.ALLOW_NO_INDICES
+            || indicesAccessControl == IndicesAccessControl.DENIED
+            || indicesAccessControl == IndicesAccessControl.allowAll()
+            || indicesAccessControl.getClass().equals(IndicesAccessControl.class) == false) {
```

Comment:
The last condition is not exactly a test for "already wrapped". It should test if the class is already an instance of `DlsFlsFeatureTrackingIndicesAccessControlWrapper`.